Mage.Sets/src/mage/cards/a/AKillerAmongUs.java

```diff
@@ -0,0 +1,236 @@
+        super(card);
+    }
+
+    @Override
+    public AKillerAmongUs copy() {
+        return new AKillerAmongUs(this);
+    }
+}
+
+class ChooseHumanMerfolkOrGoblinEffect extends OneShotEffect {
+
+    public static final String SECRET_CREATURE_TYPE = "_susCreatureType";
+    public static final String SECRET_OWNER = "_secOwn";
+
+    ChooseHumanMerfolkOrGoblinEffect() {
+        super(Outcome.Neutral);
+        staticText = "Then secretly choose Human, Merfolk, or Goblin.";
+    }
+
+    private ChooseHumanMerfolkOrGoblinEffect(final ChooseHumanMerfolkOrGoblinEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public ChooseHumanMerfolkOrGoblinEffect copy() {
+        return new ChooseHumanMerfolkOrGoblinEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null) {
+            return false;
+        }
+        Permanent permanent = game.getPermanentOrLKIBattlefield(source.getSourceId());
+        if (permanent == null) {
+            return false;
+        }
+
+        Choice choice = new ChoiceImpl();
+        Set<String> choices = new LinkedHashSet<>();
+        choices.add("Human");
+        choices.add("Merfolk");
+        choices.add("Goblin");
+        choice.setChoices(choices);
+        choice.setMessage("Choose Human, Merfolk, or Goblin");
+
+        controller.choose(outcome, choice, game);
+        game.informPlayers(permanent.getName() + ": " + controller.getLogName() + " has secretly chosen a creature type.");
+
+        SubType chosenType = SubType.fromString(choice.getChoice());
+        setSecretCreatureType(chosenType, source, game);
+        setSecretOwner(source.getControllerId(), source, game);
+        return true;
+    }
+
+    public static void setSecretCreatureType(SubType type, Ability source, Game game) {
+        game.getState().setValue(getThing(source, game) + SECRET_CREATURE_TYPE, type);
+    }
+
+    public static SubType getSecretCreatureType(Ability source, Game game) {
+        return (SubType) game.getState().getValue(getThing(source, game) +
+                ChooseHumanMerfolkOrGoblinEffect.SECRET_CREATURE_TYPE);
+    }
+
+    public static void setSecretOwner(UUID owner, Ability source, Game game) {
+        game.getState().setValue(getThing(source, game) + SECRET_OWNER, owner);
+    }
+
+    public static UUID getSecretOwner(Ability source, Game game) {
+        return (UUID) game.getState().getValue(getThing(source, game) +
+                ChooseHumanMerfolkOrGoblinEffect.SECRET_OWNER);
+    }
+
+    private static String getThing(Ability source, Game game) {
+        if (game.getPermanentOrLKIBattlefield(source.getSourceId()) != null) {
+            return "" + source.getSourceId() + "_" + (game.getPermanentOrLKIBattlefield(source.getSourceId()).getZoneChangeCounter(game));
+        }
+        return "Does not exist";
```

Comment:
this seems potentially problematic in that it's being used as a key directly, which is okay for get usage if it won't exist, but "does not exist" is definitely not something that should be set to